From: Wanpeng Li <liwanp@linux.vnet.ibm.com>
Date: Tue, 17 Dec 2013 10:45:52 +1100
Subject: [PATCH] mm/migrate.c: fix setting of cpupid on page migration twice
 against normal page

References: Automatic NUMA Balancing (fate#315482)
Patch-mainline: v3.14
Git-commit: a3978a519461b095b776f44a86079f5448c96963

7851a45cd3 ("mm: numa: Copy cpupid on page migration") copies over the
cpupid at page migration time.  it is unnecessary to set it again in
alloc_misplaced_dst_page().

Signed-off-by: Wanpeng Li <liwanp@linux.vnet.ibm.com>
Reviewed-by: Naoya Horiguchi <n-horiguchi@ah.jp.nec.com>
Acked-by: Mel Gorman <mgorman@suse.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/migrate.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mm/migrate.c b/mm/migrate.c
index 85b628f..a336478 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -1558,8 +1558,6 @@ static struct page *alloc_misplaced_dst_page(struct page *page,
 					  __GFP_NOMEMALLOC | __GFP_NORETRY |
 					  __GFP_NOWARN) &
 					 ~GFP_IOFS, 0);
-	if (newpage)
-		page_cpupid_xchg_last(newpage, page_cpupid_last(page));
 
 	return newpage;
 }
